### ISSUE TITLE
some questions about the train, val and test set

### ISSUE THREAD

**Issue by zqyJason** [ultralytics/yolov5#8256]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi, guys! As for the MS coco, I have noticed three datasets: `train, val `and` test` in this project. However, I have a question: is the `val `in the  `coco.yaml `a validation set the same concept as in machine learning? ——Participating in the training process of the network, and perform model tuning and adjustment. My personal understanding is that the `val `here is actually a test set, after each epoch of the model training, the model is tested and the process is repeated until the end of the training, right?. 

So , if I don't have a `val` set, can I just use the `test `set as the `val `set? Does this have any effect on model training? One more question, if I have three test sets (with the same distribution in the same field, with different levels of difficulty), how do I save the final trained model so that the final  test metrics on each test set (such as AP, Ap50, etc.) are relatively better?

**Comment by glenn-jocher** [comment]
@zqyJason train and val sets are required, test is not. See https://en.wikipedia.org/wiki/Training,_validation,_and_test_data_sets for details
